src/test/kotlin/no/nav/familie/ef/sak/vilkår/regler/vilkår/NyttBarnSammePartnerRegelTest.kt

```diff
@@ -0,0 +1,101 @@
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.time.LocalDate
+import java.util.*
+
+class NyttBarnSammePartnerRegelTest {
+
+    val hovedregelMetadataMock = mockk<HovedregelMetadata>()
+    val behandlingBarn = behandlingBarn(behandlingId = UUID.randomUUID())
+
+    @BeforeEach
+    fun setup() {
+        every { hovedregelMetadataMock.behandling } returns behandling()
+    }
+
+    @Test
+    fun `Gitt at bruker eller annen foreldre ikke har tidligere vedtak, og alle barn har registrert annen forelder, når initiereDelvilkår, så skal vilkåret automatisk oppfylles`() {
+        every { hovedregelMetadataMock.barn } returns listOf(behandlingBarn)
+        every { hovedregelMetadataMock.vilkårgrunnlagDto } returns VilkårTestUtil.mockVilkårGrunnlagDto()
+
+        val listDelvilkårsvurdering = NyttBarnSammePartnerRegel().initiereDelvilkårsvurdering(
+            hovedregelMetadataMock,
+            Vilkårsresultat.IKKE_TATT_STILLING_TIL,
+            null
+        )
+
+        assertThat(listDelvilkårsvurdering.size).isEqualTo(1)
+        assertThat(listDelvilkårsvurdering.first().resultat).isEqualTo(Vilkårsresultat.AUTOMATISK_OPPFYLT)
+        assertThat(listDelvilkårsvurdering.first().vurderinger.size).isEqualTo(1)
+        assertThat(listDelvilkårsvurdering.first().vurderinger.first().svar).isEqualTo(SvarId.NEI)
+        assertThat(listDelvilkårsvurdering.first().vurderinger.first().begrunnelse).startsWith("Automatisk vurdert (${LocalDate.now().norskFormat()}):")
+    }
+
+    @Test
+    fun `Gitt at bruker eller annen foreldre har tidligere vedtak, og alle barn har registrert annen forelder, når initiereDelvilkår, så skal vilkåret ikke automatisk oppfylles`() {
+        val barnMedSamværSøknadsgrunnlagDto = mockk<BarnMedSamværSøknadsgrunnlagDto>()
+        val annenForelderDto = mockk<AnnenForelderDto>()
+        every { hovedregelMetadataMock.barn } returns listOf(behandlingBarn)
+        every { hovedregelMetadataMock.vilkårgrunnlagDto } returns VilkårTestUtil.mockVilkårGrunnlagDto(
+            barnMedSamvær = listOf(BarnMedSamværDto(UUID.randomUUID(), barnMedSamværSøknadsgrunnlagDto, mockk())),
+            tidligereVedtaksperioder = TidligereVedtaksperioderDto(TidligereInnvilgetVedtakDto(true, false, false), TidligereInnvilgetVedtakDto(false, false, false), false)
+        )
+        every { annenForelderDto.tidligereVedtaksperioder } returns TidligereVedtaksperioderDto(null, null, false)
+        every { barnMedSamværSøknadsgrunnlagDto.forelder } returns annenForelderDto
+        val listDelvilkårsvurdering = NyttBarnSammePartnerRegel().initiereDelvilkårsvurdering(
+            hovedregelMetadataMock,
+            Vilkårsresultat.IKKE_TATT_STILLING_TIL,
+            null
+        )
+
+        assertThat(listDelvilkårsvurdering.size).isEqualTo(1)
+        assertThat(listDelvilkårsvurdering.first().resultat).isEqualTo(Vilkårsresultat.IKKE_TATT_STILLING_TIL)
```

Comment:
For å unngå allt for lange rader så kan det være fint å trekke ut noen deler til variabler, eks `listDelvilkårsvurdering.first()`